Discord.Net.Abstractions/Core/IGuildEmote.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Discord
+{
+    /// <inheritdoc cref="GuildEmote" />
+    public interface IGuildEmote : IEmoteEntity
+    {
+        /// <inheritdoc cref="GuildEmote.IsManaged" />
+        bool IsManaged { get; }
+
+        /// <inheritdoc cref="GuildEmote.RequireColons" />
+        bool RequireColons { get; }
+
+        /// <inheritdoc cref="GuildEmote.RoleIds" />
+        IReadOnlyList<ulong> RoleIds { get; }
+    }
+
+    /// <summary>
+    /// Provides an abstraction wrapper layer around a <see cref="Discord.GuildEmote"/>, through the <see cref="IGuildEmote"/> interface.
+    /// </summary>
+    public class GuildEmoteAbstraction : EmoteAbstraction, IGuildEmote
+    {
+        /// <summary>
+        /// Constructs a new <see cref="GuildEmoteAbstraction"/> around an existing <see cref="Discord.GuildEmote"/>.
+        /// </summary>
+        /// <param name="guildEmote">The existing <see cref="Discord.GuildEmote"/> to be abstracted.</param>
+        /// <exception cref="ArgumentNullException">Throws for <paramref name="guildEmote"/>.</exception>
+        public GuildEmoteAbstraction(GuildEmote guildEmote)
+            : base(guildEmote) { }
+
+        /// <inheritdoc />
+        public bool IsManaged
+            => GuildEmote.IsManaged;
+
+        /// <inheritdoc />
+        public bool RequireColons
+            => GuildEmote.RequireColons;
+
+        /// <inheritdoc />
+        public IReadOnlyList<ulong> RoleIds
+            => GuildEmote.RoleIds;
+
+        /// <inheritdoc cref="GuildEmote.ToString" />
+        public override string ToString()
+            => Emote.ToString();
+
+        /// <summary>
+        /// The existing <see cref="Discord.GuildEmote"/> being abstracted.
+        /// </summary>
+        internal protected GuildEmote GuildEmote
```

Comment:
This looks like the only `internal protected` D.Net property in your abstractions. Should it just be `protected` instead?

Reply:
It's because this is the only one that has an `Unabstract` extension method, which you can see if you scroll down to the bottom of that file. 

It's used in a few spots in `IRestGuild.cs` and `ISocketGuild.cs` as follows...

```
public Task DeleteEmoteAsync(IGuildEmote emote, RequestOptions options = null)
    => RestGuild.DeleteEmoteAsync(emote.Unabstract(), options);
```

That's definitely one of the more creative workarounds I've got in here that I don't really like, but as far as I can tell, there's no possible way for consumers to create an `Emote` or `GuildEmote` object, so I figured it should be abstracted as a parameter.

Reply:
Ah, ok. That works then